File: server/makc2022--dotnet/Makc2022.Layer3.Sql.Sample/Entities/DummyMain/DummyMainEntityLoader.cs
// Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

using Makc2022.Layer3.Sql.Sample.Entity;

namespace Makc2022.Layer3.Sql.Sample.Entities.DummyMain
{
    /// <summary>
    /// Загрузчик сущности "DummyMain".
    /// </summary>
    public class DummyMainEntityLoader : EntityLoader<DummyMainEntityObject>
    {
        #region Constructors

        /// <inheritdoc/>
        public DummyMainEntityLoader(DummyMainEntityObject? entityObject = null)
            : base(entityObject ?? new DummyMainEntityObject())
        {
        }

        #endregion Constructors

        #region Public methods

        /// <inheritdoc/>
        public sealed override HashSet<string> Load(
            DummyMainEntityObject entityObject,
            HashSet<string>? loadableProperties = null)
        {
            var result = base.Load(entityObject, loadableProperties);

            if (result.Contains(nameof(EntityObject.Id)))
            {
                EntityObject.Id = entityObject.Id;
            }

            if (result.Contains(nameof(EntityObject.IdOfDummyOneToManyEntity)))
            {
                EntityObject.IdOfDummyOneToManyEntity = entityObject.IdOfDummyOneToManyEntity;
            }

            if (result.Contains(nameof(EntityObject.Name)))
            {
                EntityObject.Name = entityObject.Name ?? string.Empty;
            }

            if (result.Contains(nameof(EntityObject.PropBoolean)))
            {
                EntityObject.PropBoolean = entityObject.PropBoolean;
            }

            if (result.Contains(nameof(EntityObject.PropBooleanNullable)))
            {
                EntityObject.PropBooleanNullable = entityObject.PropBooleanNullable;
            }

            if (result.Contains(nameof(EntityObject.PropDate)))
            {
                EntityObject.PropDate = entityObject.PropDate;
            }

            if (result.Contains(nameof(EntityObject.PropDateNullable)))
            {
                EntityObject.PropDateNullable = entityObject.PropDateNullable;
            }

            if (result.Contains(nameof(EntityObject.PropDateTimeOffset)))
            {
                EntityObject.PropDateTimeOffset = entityObject.PropDateTimeOffset;
            }

            if (result.Contains(nameof(EntityObject.PropDateTimeOffsetNullable)))
            {
                EntityObject.PropDateTimeOffsetNullable = entityObject.PropDateTimeOffsetNullable;
            }

            if (result.Contains(nameof(EntityObject.PropDecimal)))
            {
                EntityObject.PropDecimal = entityObject.PropDecimal;
            }

            if (result.Contains(nameof(EntityObject.PropDecimalNullable)))
            {
                EntityObject.PropDecimalNullable = entityObject.PropDecimalNullable;
            }

            if (result.Contains(nameof(EntityObject.PropInt32)))
            {
                EntityObject.PropInt32 = entityObject.PropInt32;
            }

            if (result.Contains(nameof(EntityObject.PropInt32Nullable)))
            {
                EntityObject.PropInt32Nullable = entityObject.PropInt32Nullable;
            }

            if (result.Contains(nameof(EntityObject.PropInt64)))
            {
                EntityObject.PropInt64 = entityObject.PropInt64;
            }

            if (result.Contains(nameof(EntityObject.PropInt64Nullable)))
            {
                EntityObject.PropInt64Nullable = entityObject.PropInt64Nullable;
            }

            if (result.Contains(nameof(EntityObject.PropString)))
            {
                EntityObject.PropString = entityObject.PropString ?? string.Empty;
            }

            if (result.Contains(nameof(EntityObject.PropStringNullable)))
            {
                EntityObject.PropStringNullable = entityObject.PropStringNullable;
            }

            return result;
        }

        #endregion Public methods

        #region Protected methods

        /// <inheritdoc/>
        protected override HashSet<string> CreateAllPropertiesToLoad()
        {
            return new HashSet<string>
            {
                nameof(EntityObject.Id),
                nameof(EntityObject.Name),
                nameof(EntityObject.IdOfDummyOneToManyEntity),
                nameof(EntityObject.PropBoolean),
                nameof(EntityObject.PropBooleanNullable),
                nameof(EntityObject.PropDate),
                nameof(EntityObject.PropDateNullable),
                nameof(EntityObject.PropDateTimeOffset),
                nameof(EntityObject.PropDateTimeOffsetNullable),
                nameof(EntityObject.PropDecimal),
                nameof(EntityObject.PropDecimalNullable),
                nameof(EntityObject.PropInt32),
                nameof(EntityObject.PropInt32Nullable),
                nameof(EntityObject.PropInt64),
                nameof(EntityObject.PropInt64Nullable),
                nameof(EntityObject.PropString),
                nameof(EntityObject.PropStringNullable)
            };
        }

        #endregion Protected methods
    }
}

File: server/makc2022--nodejs--nestjs/src/layer3/nosql-mongo/sample/entities/dummy-main/dummy-main-entity.loader.ts
/** Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License. */

import { nameof } from '../../../../../layer1/common/common-functions';
import { EntityLoader } from '../../../../../layer1/entity/entity-loader';
import { DummyMainEntityObject } from './dummy-main-entity.object';

/** Загрузчик сущности "Dummy". */
export class DummyMainEntityLoader extends EntityLoader<DummyMainEntityObject> {
  /** @inheritdoc */
  constructor(entityObject?: DummyMainEntityObject) {
    super(entityObject ?? ({} as DummyMainEntityObject));
  }

  /** @inheritdoc */
  override load(
    entityObject: DummyMainEntityObject,
    loadableProperties?: Set<string>
  ): Set<string> {
    const result = super.load(entityObject, loadableProperties);

    if (result.has(nameof<DummyMainEntityObject>('id'))) {
      this.entityObject.id = entityObject.id;
    }

    if (result.has(nameof<DummyMainEntityObject>('idOfDummyOneToManyEntity'))) {
      this.entityObject.idOfDummyOneToManyEntity = entityObject.idOfDummyOneToManyEntity;
    }

    if (result.has(nameof<DummyMainEntityObject>('name'))) {
      this.entityObject.name = entityObject.name;
    }

    if (result.has(nameof<DummyMainEntityObject>('propBoolean'))) {
      this.entityObject.propBoolean = entityObject.propBoolean;
    }

    if (result.has(nameof<DummyMainEntityObject>('propBooleanNullable'))) {
      this.entityObject.propBooleanNullable = entityObject.propBooleanNullable;
    }

    if (result.has(nameof<DummyMainEntityObject>('propDate'))) {
      this.entityObject.propDate = entityObject.propDate;
    }

    if (result.has(nameof<DummyMainEntityObject>('propDateNullable'))) {
      this.entityObject.propDateNullable = entityObject.propDateNullable;
    }

    if (result.has(nameof<DummyMainEntityObject>('propDecimal'))) {
      this.entityObject.propDecimal = entityObject.propDecimal;
    }

    if (result.has(nameof<DummyMainEntityObject>('propDecimalNullable'))) {
      this.entityObject.propDecimalNullable = entityObject.propDecimalNullable;
    }

    if (result.has(nameof<DummyMainEntityObject>('propInt32'))) {
      this.entityObject.propInt32 = entityObject.propInt32;
    }

    if (result.has(nameof<DummyMainEntityObject>('propInt32Nullable'))) {
      this.entityObject.propInt32Nullable = entityObject.propInt32Nullable;
    }

    if (result.has(nameof<DummyMainEntityObject>('propInt64'))) {
      this.entityObject.propInt64 = entityObject.propInt64;
    }

    if (result.has(nameof<DummyMainEntityObject>('propInt64Nullable'))) {
      this.entityObject.propInt64Nullable = entityObject.propInt64Nullable;
    }

    if (result.has(nameof<DummyMainEntityObject>('propString'))) {
      this.entityObject.propString = entityObject.propString;
    }

    if (result.has(nameof<DummyMainEntityObject>('propStringNullable'))) {
      this.entityObject.propStringNullable = entityObject.propStringNullable;
    }

    return result;
  }

  /** @inheritdoc */
  protected override createAllPropertiesToLoad(): Set<string> {
    return new Set<string>([
      nameof<DummyMainEntityObject>('id'),
      nameof<DummyMainEntityObject>('idOfDummyOneToManyEntity'),
      nameof<DummyMainEntityObject>('name'),
      nameof<DummyMainEntityObject>('propBoolean'),
      nameof<DummyMainEntityObject>('propBooleanNullable'),
      nameof<DummyMainEntityObject>('propDate'),
      nameof<DummyMainEntityObject>('propDateNullable'),
      nameof<DummyMainEntityObject>('propDecimal'),
      nameof<DummyMainEntityObject>('propDecimalNullable'),
      nameof<DummyMainEntityObject>('propInt32'),
      nameof<DummyMainEntityObject>('propInt32Nullable'),
      nameof<DummyMainEntityObject>('propInt64'),
      nameof<DummyMainEntityObject>('propInt64Nullable'),
      nameof<DummyMainEntityObject>('propString'),
      nameof<DummyMainEntityObject>('propStringNullable'),
    ]);
  }
}
